vslib/inc/MACsecFilter.h

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "TrafficFilter.h"
+
+#include <string>
+
+namespace saivs
+{
+    class MACsecFilter : public TrafficFilter
+    {
+    public:
+        MACsecFilter(
+            _In_ const std::string &macsec_interface_name);
+
+        virtual ~MACsecFilter() = default;
+
+        virtual FilterStatus execute(
+            _Inout_ void *buffer,
+            _Inout_ ssize_t &length) override;
```

Comment:
why signed size ? and not size_t ?

Reply:
This function will be call after `recvmsg`, So I directly use the type in this code
https://github.com/Azure/sonic-sairedis/blob/425ffbcbf57fdb66160c1ba84bbec2c0f1f5e983/vslib/src/HostInterfaceInfo.cpp#L145
Should I cast it to unsigned size?

Reply:
ok, this is fine, i though this size is buffer length, since then it make no sense for it to allow negative numbers
so i looked at  your code, so you actually passing &length, but forward function is without (&), and you are not modifying length parameter in any way, so it should also be without (&), then recvmsg returns ssize, and its negative when error, but you done return that ether, you have your own FilterStatus error code, so i think this ssize_t& should be change to size_t in all places and if needed cast to int in recvmgs method since thats the definition

Reply:
I agree your point. `size_t` should make more sense.
And the function `execute`, I think, is for processing the data, maybe it will modify the data. So the paramete of `execute`  are with (&).
But the function `forward`, I think, is for forwarding data. The forwarding action will never change the data. So the parameters are without (&).